domain/network/service/space.go

```diff
@@ -61,6 +62,47 @@ func NewSpaceService(st State, logger Logger) *SpaceService {
 	}
 }
 
+// AddSpace creates and returns a new space.
+func (s *SpaceService) AddSpace(ctx context.Context, name string, providerID network.Id, subnetIDs []string) (utils.UUID, error) {
```

Comment:
Add `network.UUID` to `core/network` (or `domain/network`) and return the UUID. That way we're not returning strings and secondary (most importantly), we're not passing `utils.UUID` around.

```suggestion
func (s *SpaceService) AddSpace(ctx context.Context, name string, providerID network.Id, subnetIDs []string) (network.UUID, error) {
```